drivers/usbdev/cdcacm.c

```diff
@@ -2342,7 +2342,11 @@ static int cdcuart_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
         termiosp->c_iflag = serdev->tc_iflag;
         termiosp->c_oflag = serdev->tc_oflag;
         termiosp->c_lflag = serdev->tc_lflag;
-        termiosp->c_cflag = CS8;
+        termiosp->c_cflag =
+            ((priv->linecoding.parity != CDC_PARITY_NONE) ? PARENB : 0) |
```

Comment:
```suggestion
            ((priv->linecoding.parity == CDC_PARITY_EVEN) ? PARENB : 0) |
```

Reply:
@xiaoxiang781216 Is it the double negative you do not like? 
PARENB is enable.  With your suggestion, would it be enabled on Odd parity?

Reply:
both CDC_PARITY_EVEN and CDC_PARITY_ODD will set If priv->linecoding.parity == CDC_PARITY_ODD, I don't think it's the right output.

Reply:
> both CDC_PARITY_EVEN and CDC_PARITY_ODD will set If priv->linecoding.parity == CDC_PARITY_ODD, I don't think it's the right output.

I do not understand what you typed. May be you could rephrase it?

Here is how I understand this:

CDC settings
```
#define CDC_PARITY_NONE         0 /* No parity  */
#define CDC_PARITY_ODD          1 /* Odd parity */
#define CDC_PARITY_EVEN         2 /* Even parity */
#define CDC_PARITY_MARK         3 /* Mark parity */
#define CDC_PARITY_SPACE        4 /* Space parity */
```
We do not support mark and space parity

Termios Settings:
```
#define PARENB    (1 << 8)    /* Bit 8: Parity enable */
#define PARODD    (1 << 9)    /* Bit 9: Odd parity, else even */
```

At the logical level:

PARENB (FALSE) when v = CDC_PARITY_NONE     
PARENB (TRUE)  when v = CDC_PARITY_ODD  OR CDC_PARITY_EVEN  (short circuited as NOT CDC_PARITY_NONE)
PARODD (FALSE) when v = CDC_PARITY_EVEN  OR CDC_PARITY_NONE 
PARODD (FALSE) when v = CDC_PARITY_ODD

|PARENB |  PARODD |                         |                                        |
| ---------|------------|-----------------| --------------------------| 
|   0         |    0           | No Parity          |   CDC_PARITY_NONE     |             
|   1         |    1            | Odd Parity        |   CDC_PARITY_ODD      |
|   1         |     0            | Even Parity       |   CDC_PARITY_EVEN      |   

@xiaoxiang781216 - The code matches the logic.

Reply:
@davids5 sorry, I misunderstand the PARENB meaning(enable parity check, not even parity check).

Reply:
@xiaoxiang781216  - OK - please merge it.